### PR TITLE
[Modules] Updated Automation/AutomationAccounts to new dependency approach

### PR DESCRIPTION
**Pull request by MrMCake**
# Description

- Updated Automation/AutomationAccounts to new dependency approach

| Pipeline |
| - |
| [Automation: AutomationAccounts](https://github.com/Azure/ResourceModules/actions/workflows/ms.automation.automationaccounts.yml) |

# Type of Change

Please delete options that are not relevant.

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Update to documentation

### CODE REVIEW

File: modules/Microsoft.Automation/automationAccounts/.test/common/deploy.test.bicep [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+        privateDnsZoneGroups: {
+          privateDNSResourceIds: [
+            resourceGroupResources.outputs.privateDNSZoneResourceId
+          ]
+        }
+        service: 'Webhook'
+        subnetResourceId: resourceGroupResources.outputs.subnetResourceId
+      }
+      {
+        privateDnsZoneGroups: {
```
Comment (eriqua) [inline modules/Microsoft.Automation/automationAccounts/.test/common/deploy.test.bicep:99]:
```suggestion
        privateDnsZoneGroup: {
```

File: modules/Microsoft.Automation/automationAccounts/.test/common/deploy.test.bicep [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+    modules: [
+      {
+        name: 'PSWindowsUpdate'
+        uri: 'https://www.powershellgallery.com/api/v2/package'
+        version: 'latest'
+      }
+    ]
+    privateEndpoints: [
+      {
+        privateDnsZoneGroups: {
```
Comment (eriqua) [inline modules/Microsoft.Automation/automationAccounts/.test/common/deploy.test.bicep:90]: privateDnsZoneGroup

Comment (MrMCake) [reply modules/Microsoft.Automation/automationAccounts/.test/common/deploy.test.bicep]: updated

File: modules/Microsoft.Automation/automationAccounts/.test/encr/deploy.test.bicep [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+module testDeployment '../../deploy.bicep' = {
+  scope: resourceGroup
+  name: '${uniqueString(deployment().name)}-test-${serviceShort}'
+  params: {
+    name: '<<namePrefix>>${serviceShort}001'
+    cMKKeyName: resourceGroupResources.outputs.keyVaultEncryptionKeyName
+    cMKKeyVaultResourceId: resourceGroupResources.outputs.keyVaultResourceId
+    cMKUserAssignedIdentityResourceId: resourceGroupResources.outputs.managedIdentityResourceId
+    publicNetworkAccess: 'Enabled'
```
Comment (eriqua) [inline modules/Microsoft.Automation/automationAccounts/.test/encr/deploy.test.bicep:52]: That should get enabled by default if pe are not set. Should we remove this line?

Comment (MrMCake) [reply modules/Microsoft.Automation/automationAccounts/.test/encr/deploy.test.bicep:52]: sure, why not.

**Review by eriqua — APPROVED** [review]
🆗

**Pull request merged**